### PR TITLE
Add magazine wells to items

### PR DESCRIPTION
**Pull request by ymber**
#### Summary
SUMMARY: Infrastructure "Add mag wells to all relevant items"

#### Purpose of change
#40399 needed it.

#### Describe the solution
Add `MAGAZINE_WELL` type pockets to relevant items.

#### Describe alternatives you've considered
Setting appropriate maximum values for volume and weight.

#### Testing
Won't work without changes in #40399.

### CODE REVIEW

**Comment by KorGgenT** [comment]
I haven't had a chance to review it yet, but this probably addresses the comment i made in #40399

File: data/json/items/gun/50.json [hunk 1]
```diff
@@ -94,7 +109,15 @@
...
     },
     "flags": [ "RELOAD_EJECT" ],
-    "pocket_data": [ { "pocket_type": "MAGAZINE", "ammo_restriction": { "50": 1 } } ]
+    "pocket_data": [
+      {
+        "pocket_type": "MAGAZINE_WELL",
+        "holster": true,
+        "max_contains_volume": "20 L",
+        "max_contains_weight": "20 kg",
+        "item_restriction": [  ]
```
Comment (KorGgenT) [inline data/json/items/gun/50.json:118]:
```suggestion
        "ammo_restriction": { "50": 1 },
```

File: data/json/items/gun/50.json [hunk 2]
```diff
@@ -94,7 +109,15 @@
       "barrel_length": -1
     },
     "flags": [ "RELOAD_EJECT" ],
-    "pocket_data": [ { "pocket_type": "MAGAZINE", "ammo_restriction": { "50": 1 } } ]
+    "pocket_data": [
+      {
+        "pocket_type": "MAGAZINE_WELL",
```
Comment (KorGgenT) [inline data/json/items/gun/50.json:114]:
```suggestion
        "pocket_type": "MAGAZINE",
```
Related to the other comment

File: data/json/items/gun/50.json [hunk 3]
```diff
@@ -94,7 +109,13 @@
       "barrel_length": -1
     },
     "flags": [ "RELOAD_EJECT" ],
-    "pocket_data": [ { "pocket_type": "MAGAZINE", "ammo_restriction": { "50": 1 } } ]
+    "pocket_data": [
+      {
+        "pocket_type": "MAGAZINE",
+        "holster": true,
+        "ammo_restriction": { "50": 1 },
```
Comment (ZhilkinSerg) [inline data/json/items/gun/50.json:116]:
```suggestion
        "ammo_restriction": { "50": 1 }
```

File: data/json/items/gun/38.json [hunk 4]
```diff
@@ -90,8 +90,15 @@
...
     "clip_size": 6,
-    "magazines": [ [ "38", [ "38_speedloader6" ] ] ],
-    "pocket_data": [ { "pocket_type": "MAGAZINE", "ammo_restriction": { "38": 6 } } ]
+    "pocket_data": [
+      {
+        "pocket_type": "MAGAZINE_WELL",
+        "holster": true,
+        "max_contains_volume": "20 L",
+        "max_contains_weight": "20 kg",
+        "item_restriction": [ "38_speedloader6" ]
```
Comment (KorGgenT) [inline data/json/items/gun/38.json:99]: so all of the guns with integral magazines that can be used with speedloaders should still have an integral magazine, not a magazine well - in the reloading code it just allows you to use a speedloader to put the ammo into the gun, so it should be a MAGAZINE type

**Pull request merged**